ciphey/basemods/Decoders/dna.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional, Dict, List
+
+from ciphey.iface import Config, ParamSpec, T, U, Decoder, registry, Translation
+
+from loguru import logger
+
+import re
+
+
+@registry.register
+class Dna(Decoder[str, str]):
+    def decode(self, ctext: T) -> Optional[U]:
+        """
+        Performs DNA decoding
+        """
+        logger.trace("Attempting DNA decoder")
+        ctext_decoded = ""
+        ctext = re.sub(r"[,;:\-\s]", "", ctext)
+        ctext = " ".join(ctext[i : i + 3] for i in range(0, len(ctext), 3))
+        ctext_split = ctext.split(" ")
+
+        try:
+            for i in ctext_split:
+                ctext_decoded += self.DNA_DICT[i]
+            logger.debug(f"DNA decoded string is {ctext_decoded}")
+            return ctext_decoded
+        except Exception:
+            return None
+
+    @staticmethod
+    def priority() -> float:
+        return 0.2
```

Comment:
0.2 is very very high, I have never seen DNA in a CTF before :P

Reply:
Yeah I know. I think the reason why I did that, was it wasn't working when I put it lower. Honestly this whole priority system isn't that good in my opinion.

Reply:
Yup! It sucks, it'll be fixed in the next searcher update. Right after tagging and new checkers :sweat_smile: